be/test/vec/columns/column_array_test.cpp

```diff
@@ -0,0 +1,425 @@
+        auto column = check_and_get_column<ColumnArray>(
+                remove_nullable(array_columns[i]->assume_mutable()).get());
+        auto check_type = remove_nullable(array_types[i]);
+        Field a;
+        column->get(column->size() - 1, a);
+        Array af = a.get<Array>();
+        if (af.size() > 0) {
+            Field ef = af[0];
+            for (int j = 0; j < max_array_size_as_field; ++j) {
+                af.push_back(ef);
+            }
+            std::cout << "array size: " << af.size() << std::endl;
+            auto cloned = column->clone_resized(0);
+            cloned->insert(af);
+            std::cout << "cloned size: " << cloned->size() << std::endl;
+            // get cloned offset size
+            auto cloned_offset_size =
+                    check_and_get_column<ColumnArray>(cloned.get())->get_offsets().back();
+            std::cout << "cloned offset size: " << cloned_offset_size << std::endl;
+
+            Field f;
+            // test get
+            EXPECT_ANY_THROW({ cloned->get(0, f); });
+            // test operator[]
+            EXPECT_ANY_THROW({ cloned->operator[](0); });
+        }
+    }
+}
+
+TEST_F(ColumnArrayTest, IsDefaultAtTest) {
+    // default means meet empty array row in column_array, now just only used in ColumnObject.
+    // test is_default_at
+    for (int i = 0; i < array_columns.size(); i++) {
+        auto column = check_and_get_column<ColumnArray>(
+                remove_nullable(array_columns[i]->assume_mutable()).get());
+        auto column_size = column->size();
+        for (int j = 0; j < column_size; j++) {
+            auto is_default = column->is_default_at(j);
+            if (is_default) {
+                // check field Array is empty
+                Field f;
+                column->get(j, f);
+                auto array = f.get<Array>();
+                EXPECT_EQ(array.size(), 0) << "array is not empty";
+            }
+        }
+    }
+}
+
+TEST_F(ColumnArrayTest, HasEqualOffsetsTest) {
+    // test has_equal_offsets which more likely used in function, eg: function_array_zip
+    for (int i = 0; i < array_columns.size(); i++) {
+        auto column = check_and_get_column<ColumnArray>(
+                remove_nullable(array_columns[i]->assume_mutable()).get());
+        auto cloned = array_columns[i]->clone_resized(array_columns[i]->size());
+        auto cloned_arr =
+                check_and_get_column<ColumnArray>(remove_nullable(cloned->assume_mutable()).get());
+        // test expect true
+        EXPECT_EQ(column->get_offsets().size(), cloned_arr->get_offsets().size());
+        EXPECT_TRUE(column->has_equal_offsets(*cloned_arr));
+        // cloned column size is not equal to column size
+        cloned->pop_back(1);
+        EXPECT_FALSE(column->has_equal_offsets(*cloned_arr));
+        cloned->insert_default();
+        EXPECT_FALSE(column->has_equal_offsets(*cloned_arr));
+    }
+}
+
+} // namespace doris::vectorized
```

Comment:
There are 26 common and array special test cases. It seems to be NOT enough? How many interface are covered and not covered?